platform/windows/windows_terminal_logger.cpp

```diff
@@ -42,20 +42,30 @@ void WindowsTerminalLogger::logv(const char *p_format, va_list p_list, bool p_er
 		return;
 	}
 
-	const unsigned int BUFFER_SIZE = 16384;
-	char buf[BUFFER_SIZE + 1]; // +1 for the terminating character
-	int len = vsnprintf(buf, BUFFER_SIZE, p_format, p_list);
-	if (len <= 0) {
-		return;
+	const int static_buffer_size = 1024;
+	char static_buf[static_buffer_size];
+	char *buf = static_buf;
+	va_list list_copy;
+	va_copy(list_copy, p_list);
+	int len = vsnprintf(buf, static_buffer_size, p_format, p_list);
+	if (len >= static_buffer_size) {
+		buf = (char *)memalloc(len + 1);
+		len = vsnprintf(buf, len + 1, p_format, list_copy);
+	}
+	va_end(list_copy);
+
+	String str_buf = String::utf8(buf, len).replace("\r\n", "\n").replace("\n", "\r\n");
```

Comment:
```suggestion
	String str_buf = String::utf8(buf, len).replace("\n", "\r\n");
```

Reply:
String can already have `\r\n`s and without removing it first it will result in `\r\r\n`.

Reply:
> String can already have `\r\n`s and without removing it first it will result in `\r\r\n`.

Makes sense👍